### ISSUE TITLE
Watir waits for default_wait when checking for non-existence of an element that was not located previously.

### ISSUE THREAD

**Issue by Castone22** [watir/watir#827]
## Meta -
Watir Version: 6.15.1
Selenium Version: 3.141.0
Browser Version: Chrome 71.0.3578.80
Browser Driver Version: ChromeDriver 2.44.609538 (b655c5a60b0b544917107a59d4153d4bf78e1b90)
OS Version: Windows 10 Enterprise 1709-16299.785

## Expected Behavior -
Element existence and presence, AND lack of existence, or presence should be determined quickly so that tests may execute quickly.

Often we need to determine if an element is NOT present on a page, say, we approached a reused template from a different route though out application and a couple of text fields have not been added to the DOM.

## Actual Behavior -
Currently existence resolves nearly instantly element.present?, element.wait_until(timeout:0, &:present?) and such behave as one might expect.  The problem comes into play when we're checking !element.present? for any particular reason, and the element is checked for with a 30 second wait time as watir attempts to locate it. 

![image](https://user-images.githubusercontent.com/12468749/49748045-bcdef600-fc72-11e8-90f3-cf9b3a78e23d.png)
As we dive down the pictured stack, we can see that watir is calling locate into wait_for_exists which calls wait_until(&:exists?) without a timeout argument forcing watir to wait for 30 seconds for our expected non-existent element.

There currently exists a work around, where you can define a block like
```ruby  
  def without_timeout
    old_timeout = Watir.default_timeout
    Watir.default_timeout = 0
    result = yield
    Watir.default_timeout = old_timeout
    result
  end
 ```
Though this doesn't seem particularly ideal to me.

## Steps to reproduce -
navigate to any web page
search for an element that does not exist on the page using present?
observe that it takes 30 seconds to determine the element existed
try
```ruby
begin
  element.wait_until(timeout: 0, present?)
rescue Watir::Wait::TimeoutError => e
  true
end
```
And note that even though we specified a timeout watir still waits for 30 seconds.

**Comment by jkotests** [comment]
@Castone22 , how have you defined `element`? 

I have only been able to reproduce the issue when `element` is a nested element. For example:

~~~~~~~~
browser.element(id: 'does_not_exist').element.present?
#=> waits 30 seconds

browser.element(id: 'does_not_exist').present?
#=> returns immediately
~~~~~~~~

**Comment by titusfortner** [comment]
The only way into the `#wait_for` method is via `#element_call`, and only if the parameter is not `:assert_exists`. That means that the following methods will not ever hit `#wait_for`: 
1. `#exists?`
2. `#visible?`
3. `#enabled?`
4. `#present?`

Now, for nested elements... We just added an automatic waiting for `#wd` to fix #813 which is causing this... I knew there must have been a reason I didn't include that in the first place. I guess this should get put back to previous behavior, and add the waiting to the `#execute_script` methods after all. Also need to add this as a spec.

**Comment by Castone22** [comment]
We do frequently nest elements, usually similar to what you put @jkotests in functionality, in fact pretty much everything we create gets nested behind the scenes.

Seems like it would be possible to maintain the existing fix for #813 if the timeout got persisted somehow in that chain.
